src/main/java/com/google/songgame/data/TitleFormatter.java

```diff
@@ -0,0 +1,77 @@
+package com.google.songgame.data;
+
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+import java.util.regex.MatchResult;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public final class TitleFormatter {
+
+  public static String formatVideoTitle(String title) {
+    String result = title.toLowerCase();
+    result = removeInvalidParenthesesGroups(result);
+    result = removeAllSquareBracketGroups(result);
+    result = removeArtistPrefix(result);
+    result = removeExcessWhiteSpace(result);
+
+    return result;
+  }
+
+  private static String removeInvalidParenthesesGroups(String s) {
+    Pattern pattern = Pattern.compile("\\(([^\\)]+)\\)");
+    Matcher matcher = pattern.matcher(s);
+    while (matcher.find()) {
+      String currGroup = matcher.group(1);
+      // if group inside of parentheses contains invalid word, delete it from string
+      if (checkIfParenthesesGroupInvalid(currGroup)) {
+        s = s.replace("(" + currGroup + ")", "");
+        matcher = pattern.matcher(s);
+      } else {
+        // otherwise, start checking after the valid parentheses group
+        matcher.region(matcher.end(), s.length());
+      }
+    }
+    return s;
+  }
+
+  private static boolean checkIfParenthesesGroupInvalid(String s) {
+    ArrayList<String> invalidWords =
+        new ArrayList<String>(
+            Arrays.asList(
+                "feat",
+                "remix",
+                "ft",
+                "music video",
+                "official video",
+                "lyric video",
+                "official audio"));
+    for (String invalidWord : invalidWords) {
```

Comment:
Advanced and optional suggestion:

You could condense this loop using functional style Java Streams here:
```
return invalidWords.anyMatch(word -> s.contains(word));
```

This is usually how we would do the check you're doing here in Production code. Again, this is entirely option, but feel free to take a look at the following references if you want to learn more:
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#anyMatch-java.util.function.Predicate-
https://www.baeldung.com/java-8-streams-introduction#1-iterating